shopfloor_packing_info/hooks.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2021 Camptocamp SA (https://www.camptocamp.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+# The field shopfloor_packing_info on res.partner contained a text
+# that, although not necessarily equal for all the customers, it is
+# expected that many of them share similar, or equal, messages. For
+# this, a new model was declared having as purpose storing these
+# different messages, so that customers can select the one that best
+# suits them, or create a new one that can be reused for another
+# customer.
+#     This model was extracted from the original module 'shopfloor'
+# and placed in this new module, 'shopfloor_packing_info'. The idea
+# is to copy the contents of the existing column to the new model,
+# and link the new record to the customer having as content the one
+# used to create the new record. The following hooks deal with this
+# process:
+#     1. pre-init will copy, in a temporary column, the contents
+#        of the column that is going to be removed when the update
+#        on 'shopfloor' is done. Since updating & installing at the
+#        same time sometimes yields to problems in databases having
+#        already installed the module that has the field to be removed
+#        defined in some views, it deletes it from the views defined
+#        in the original module.
+#     2. post-init will get the values from that temporary column,
+#        (because the update would have removed the original column)
+#        and use them to populate the new records for the new model,
+#        and link them to the corresponding customers. The temporal
+#        column will be removed afterwards.
+
+
+import hashlib
+import os
+
+from odoo.api import SUPERUSER_ID, Environment
+
+
+def pre_init_hook(cr):
+    if _column_exists(cr, "res_partner", "shopfloor_packing_info"):
+        cr.execute(
+            """
+            ALTER TABLE res_partner
+            ADD COLUMN shopfloor_packing_info_tmp TEXT;
+
+            UPDATE res_partner
+            SET shopfloor_packing_info_tmp=shopfloor_packing_info;"""
+        )
+        cr.execute(
+            """
+            DELETE FROM ir_ui_view WHERE id IN (
```

Comment:
is this needed? The view will be already deleted by the upgrade of shopfloor, no?

Reply:
Shouldn't be needed. I'm going to remove it from here and place it in our own module, because in Travis I'm getting that error that says that a view still has a field that was removed (because of the copy of the view that is stored in the database). I thought that placing it here the problem would be fixed...